doc/format.rst

```diff
@@ -562,10 +565,12 @@ section, may be used to describe non-method attributes of the class::
   Attributes
   ----------
   x : float
-      The X coordinate.
+      Description of attribute `x`.
   y : float
-      The Y coordinate.
+      Description of attribute `y`.
 
+When referring to an attribute in the description field or elsewhere within
```

Comment:
I see. I would prefer not to remove these entirely. 

Is it within numpydoc's purview to recommend that these be rendered as links?

If so, do I understand correctly that typical options will be *just*, single backticks, single backticks preceded by `:attr:`, or single backticks preceded by some other role, and that the choice is up to the project?

Reply:
The single-backquote-around-a-parameter concept is part of the original numpydoc specification, though, so I'm not sure we can steer clear of it.

Not sure what to say about projects that reconfigure the default role. Technically, their docstrings are no longer numpydoc compliant, but it's not a big deal as long as they consistently use whichever roles they do.

Reply:
Thanks @stefanv for bringing this up again.

@rossbar can you suggest a compromise here? I think we need to address methods and attributes in some capacity. What *can* we say about them?

Reply:
> The single-backquote-around-a-parameter concept is part of the original numpydoc specification, though, so I'm not sure we can steer clear of it.

Yes, but this is only for the function parameters - recommending that class attributes and methods also be wrapped in single backticks relates to sphinx, not numpydoc, and is nowhere currently mentioned in the numpydoc standard. Generally speaking it's a fine thing to recommend, and if this were a sphinx tutorial I would be +1 for adding it - I'm only leery because this document "defines" the numpydoc docstring standard, and as written this would likely introduce more broken links in documentation (nor does it relate to the "function parameters should be in single backticks" recommendation that has been around since the beginning).

> Not sure what to say about projects that reconfigure the default role. Technically, their docstrings are no longer numpydoc compliant

That's not strictly true - the numpydoc standard doesn't prescribe how projects configure the `default_role` and there are certainly cases where projects may want to override it. To make matters even more confusing, the `default_role` configuration value [can be overridden with the `.. default-role::` directive on a per-file basis](https://docutils.sourceforge.io/docs/ref/rst/directives.html#default-role). The *only* place where there's currently a collision is the long-standing "function parameters should be in single backticks" recommendation. Currently, numpydoc recommends this but does not implement anything specific for it, so the rendered docs end up using the `default_role`. The real solution here is to fix this in numpydoc - see #484.

> can you suggest a compromise here? I think we need to address methods and attributes in some capacity. What can we say about them?

Since this is generally good advice, I'd be perfectly happy to put it in with softer wording; i.e. something that is very clearly a suggestion and not a necessary part of docstring formatting. Perhaps a less wordy version of:

```
When combined with `sphinx.ext.autodoc` or `sphinx.ext.autosummary`, included
{attributes/methods} become valid link targets and can be referenced elsewhere in the
docstring with single backticks. See <link to autoclass> for details.
```

This is hopefully clear, concise, and should do the job for ~95% of use-cases in practice. I'd personally be inclined to include a footnote after "single backticks" to add an additional note about the default role, e.g.

```
docstring with single backticks[^1]. See <link to autoclass> for details.

[^1]: Depending on the configuration of the `default_role <link to default role>`_ for your project.
```

 but it's not necessary.

Reply:
:+1: My expectation is only for backticks, anywhere in the docstring, to link to parameter names. It looks like we are in agreement on that, and I'm fine with whatever else is added as Sphinx guidance.

Reply:
> "function parameters should be in single backticks"

Is that documented, though? I see "Enclose _variables_ in single backticks" (emphasis mine). That is in the Parameters section, but I thought that if "variables" were limited to "function parameters" it would say so. The ambiguity there is what led me to try to clarify.

> `See <link to autoclass> for details.`
Is this what you have in mind?
https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#directive-autoclass

> `Depending on the configuration of the `default_role <link to default role>`_ for your project.`
This?
https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-default_role

Reply:
My best guess is that the intent was to state "function parameters", but 2008 is a long time ago.

Reply:
> Why does numpydoc make a recommendation about single backticks around parameters if it doesn't do anything with them?

Yeah this is likely an accident of fate - single backticks has always changed the formatting in the rendered docs (even if it was only to the formatting of broken rst links)